metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -668,6 +675,40 @@ object MetalsEnrichments
     def asTextEdit: Option[l.TextEdit] = {
       decodeJson(d.getData, classOf[l.TextEdit])
     }
+
+    /**
+     * Useful for decoded the diagnostic data since there are overlapping
+     * unrequired keys in the structure that causes issues when we try to
+     * deserialize the old top level text edit vs the newly nested actions.
+     */
+    object DiagnosticDataDeserializer
```

Comment:
I saw your comment in https://github.com/scalameta/metals/pull/5173#discussion_r1182779915 @tgodzik , but I still couldn't figure out a good way to do this now that we're using `b.ScalaDiagnostic` that doesn't mark it as `NotNull`. This doesn't seem to work, but if you have a better solution I'm all ears.

Reply:
Ok, so there is no sensible way around it. I found a Kotlin article that suggest using reflection for automatic detection of null fields, but let's not do that :sweat_smile: